Sources/Target/Common/ThreadBase.cpp

```diff
@@ -30,5 +30,17 @@ ErrorCode ThreadBase::modifyRegisters(
   action(state);
   return writeCPUState(state);
 }
+
+ErrorCode ThreadBase::beforeResume() {
+  if (!_process->isAlive())
+    return kErrorProcessNotFound;
+
+  BreakpointManager *bpm = _process->hardwareBreakpointManager();
+  if (bpm != nullptr) {
+    bpm->enable();
```

Comment:
Because you're doing useless work (repetitive) because of arguably bad
design.

This is not really a performance critical part of ds2 (thread creation
requires some amount of work anyway) but we should still strive for
limiting the amount of unnecessary work we do.

On Tuesday, August 9, 2016, Francis Ricci <notifications@github.com> wrote:

> In Sources/Target/Common/ThreadBase.cpp
> <https://github.com/facebook/ds2/pull/243#discussion_r74100984>:
>
> > @@ -30,5 +30,17 @@ ErrorCode ThreadBase::modifyRegisters(
> >    action(state);
> >    return writeCPUState(state);
> >  }
> > +
> > +ErrorCode ThreadBase::beforeResume() {
> > +  if (!_process->isAlive())
> > +    return kErrorProcessNotFound;
> > +
> > +  BreakpointManager *bpm = _process->hardwareBreakpointManager();
> > +  if (bpm != nullptr) {
> > +    bpm->enable();
>
> Why not? If a watchpoint is enabled on any thread, it should always be
> enabled on every thread. There's no case where a watchpoint should only be
> enabled on one thread, so why allow that from the API?
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/facebook/ds2/pull/243/files/2a73b10db4691d10fd51d07ec46fbded7b36f6af..b32f1128fa8ceb0010cf5f79262d062aca477162#r74100984>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AADHbF_3oVHADQBxHmyaEj0I7JmwdkI5ks5qeLQngaJpZM4H9FRy>
> .
>


-- 
Stephane Sezer